lib/test_cleanup.rb

```diff
@@ -104,11 +106,19 @@ def cleanup_shared
 
     private
 
-    def checkout_branch_if_needed(repository, branch = "master")
+    def default_origin_ref(repository)
+      default_branch = Utils.popen_read(
+        git, "-C", repository, "symbolic-ref", "refs/remotes/origin/HEAD", "--short"
+      ).presence ||= "origin/master"
+      default_branch.strip
+    end
+
+    def checkout_branch_if_needed(repository)
+      default_branch = default_origin_ref(repository).split("/", 2).last
```

Comment:
```suggestion
      default_branch = default_origin_ref(repository).split("/").last
```

Reply:
I did it this way because you can have "/" in branch names, which is confusing but it's the way it is

Reply:
Cool. Can you leave a comment explaining this?

Reply:
Sure!